src/reducer/cartReducer.js

```diff
@@ -0,0 +1,31 @@
+import { ADD_ITEM_TO_CART, REMOVE_ITEM_FROM_CART } from '../actions/types';
+
+const initialState = []
+
+const addToCart = (cart, product) => {
+  const cartItem = cart.find(c => c.id === product.id);
+  
+  // add 'quantity' property to new item
+  if(!product.quantity) {
+    product['quantity'] = 1;
+  }
+  
+  if(cartItem) {
+    cartItem.quantity++;
+  } else {
+    cart.push(product);
+  }
+}
+
+export default (state = initialState, action) => {
+  switch(action.type) {
+    case ADD_ITEM_TO_CART:
+      let updatedCart = state;
```

Comment:
do not use direct assignment with the state - always use spread operator. (or clone the state before assigning)